ingestion/src/metadata/ingestion/source/database/processor.py

```diff
@@ -0,0 +1,197 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Processor util to fetch pii sensitive columns
+"""
+import logging
+import re
+from abc import ABC, abstractmethod
+from enum import Enum, auto
+from typing import List, Optional
+
+from commonregex import CommonRegex
+from pydantic import BaseModel
+
+from metadata.generated.schema.api.data.createTable import CreateTableRequest
+from metadata.generated.schema.type.tagLabel import (
+    LabelType,
+    State,
+    TagLabel,
+    TagSource,
+)
+from metadata.ingestion.api.processor import Processor, ProcessorStatus
+from metadata.ingestion.ometa.ometa_api import OpenMetadata, OpenMetadataConnection
+
+PII = "PII"
+
+
+class PiiTypes(Enum):
+    """PiiTypes enumerates the different types of PII data"""
+
+    NONE = auto()
+    UNSUPPORTED = auto()
+    PHONE = auto()
+    EMAIL = auto()
+    CREDIT_CARD = auto()
+    ADDRESS = auto()
+    ADDRESS_LOCATION = auto()
+    PERSON = auto()
+    LOCATION = auto()
+    BIRTH_DATE = auto()
+    GENDER = auto()
+    NATIONALITY = auto()
+    IP_ADDRESS = auto()
+    SSN = auto()
+    USER_NAME = auto()
+    PASSWORD = auto()
+    ETHNICITY = auto()
+    TAX_ID = auto()
+    KEY = auto()
+    BANKACC = auto()
+
+
+class TagType(Enum):
+    SENSITIVE = "Sensitive"
+    NONSENSITIVE = "NonSensitive"
+
+
+class ColumnPIIType(BaseModel):
+    pii_types: PiiTypes
+    tag_type: TagType
+
+
+class Scanner(ABC):
+    @abstractmethod
+    def scan(self, text):
+        """scan the text and return array of PiiTypes that are found"""
+
+
+class RegexScanner(Scanner):
+    """A scanner that uses commmon regular expressions to find PII"""
+
+    def scan(self, text):
+        """Scan the text and return an array of PiiTypes that are found"""
+        regex_result = CommonRegex(text)
+        types = []
+        if regex_result.phones:  # pylint: disable=no-member
+            types.append(PiiTypes.PHONE.name)
+        if regex_result.emails:  # pylint: disable=no-member
+            types.append(PiiTypes.EMAIL.name)
+        if regex_result.credit_cards:  # pylint: disable=no-member
+            types.append(PiiTypes.CREDIT_CARD.name)
+        if regex_result.street_addresses:  # pylint: disable=no-member
+            types.append(PiiTypes.ADDRESS.name)
+
+        return types
+
+
+class ColumnNameScanner(Scanner):
```

Comment:
can you please add some unit tests around this? Would be interesting to have a rather big battery of names and see how they are getting flagged. This way, once a user shares with us something that is not working as expected, we add it to the validations, fix it and move on.
Thanks!

Reply:
Working on it.